### PR TITLE
Migrate /channel e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by lucassabreu**
#### Summary

Convert following support files: 
1. e2e/cypress/tests/support/api/user.js
1. e2e/cypress/tests/support/api/role.js

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/21299

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by furqanmlk** [comment]
@lucassabreu 
Can you please fix lint errors but running `make check-style` from `/mattermost-webapp` directory

**Comment by lucassabreu** [comment]
hi @furqanmlk , fixed the lint errors.

**Review by furqanmlk — COMMENTED** [review]
Thanks @lucassabreu 

Can you please add types of variables in `describe` block in migrated test cases

```
describe('Leave an archived channel', () => { 
     let testTeam: Cypress.Team;
     let offTopicUrl: string;
```

```et testTeam: Cypress.Team;
    let testChannel: Cypress.Channel;
    let groupChannel: Cypress.Channel;
    let directChannel: Cypress.Channel;
    let directUser: Cypress.UserProfile;
    let admin: Cypress.UserProfile;
    let user: Cypress.UserProfile;
```

test `e2e/cypress/tests/integration/channel/channel_info_rhs_spec.ts` is failing

There are other tests are failing, can you please check locally ?

**Comment by lucassabreu** [comment]
i run the e2e tests for channels, all passed

**Comment by lucassabreu** [comment]
just to report, i saw the failing tests at: https://automation-dashboard.vercel.app/cycles/28573f72-dbad-44b8-b924-50141b4c3292

and i am trying to fix them.

**Comment by furqanmlk** [comment]
> just to report, i saw the failing tests at: https://automation-dashboard.vercel.app/cycles/28573f72-dbad-44b8-b924-50141b4c3292
> 
> 
> 
> and i am trying to fix them.

Thank you so much @lucassabreu

**Comment by furqanmlk** [comment]
@lucassabreu please re-assign when you are done with changes.
Thanks

**Comment by lucassabreu** [comment]
hi @furqanmlk 

running locally these two did not fail, so i don't know if i need to fix anything...

tests/integration/accessibility/accessibility_account_settings_spec.js tests/integration/enterprise/system_console/about/starter_edition_spec.js

this one is broken in the server... do i need to open a PR to the other repo?
https://github.com/mattermost/mattermost-webapp/blob/a3afcdd881e8481ca4d69802d0bb42971d64e9f1/e2e/cypress/tests/integration/modals/quick_switcher_spec.js#L106

**Comment by lucassabreu** [comment]
hi @furqanmlk i merged the changes from master into the PR again and looked into the last e2e-test run and there are a lot of "skippeds", there is something that i need to do?

https://automation-dashboard.vercel.app/cycles/bc9a533c-457a-4285-859d-d10188afa387

i also saw that some of the fixes that i did are on master now...

**Comment by furqanmlk** [comment]
> hi @furqanmlk i merged the changes from master into the PR again and looked into the last e2e-test run and there are a lot of "skippeds", there is something that i need to do?
> 
> https://automation-dashboard.vercel.app/cycles/bc9a533c-457a-4285-859d-d10188afa387
> 
> i also saw that some of the fixes that i did are on master now...

Sorry @lucassabreu 
I got stuck on other tasks.
Yeah, you might see fixes on support files as other contributors also migrating tests.
The changes you are noticing are only new text, that has recently be updated. I think your fine.
Can you please resolve the conflict?

**Comment by furqanmlk** [comment]
@lucassabreu sorry for the delay 
can you please fix conflict ?
I would like to close this PR ASAP :)

**Comment by lucassabreu** [comment]
done @furqanmlk

File: components/channel_info_rhs/channel_info_rhs.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const ChannelInfoRhs = ({
     actions,
 }: Props) => {
     const currentUserId = currentUser.id;
-    const channelURL = getSiteURL() + '/' + currentTeam.name + '/channels/' + channel.name;
+    const channelURL = getSiteURL() + '/' + currentTeam?.name + '/channels/' + channel.name;
```
Comment (M-ZubairAhmed) [inline components/channel_info_rhs/channel_info_rhs.tsx:86]: why is this change needed?

Comment (lucassabreu) [reply components/channel_info_rhs/channel_info_rhs.tsx:86]: when mattermost is coming back from the system config views the currentTeam is not loaded yeat, so it is undefined.

if i remove this react will break completely and not render anything because of that.

with this it will resolve itself in the second try reading from the url, at least that was as much as i could gather from debugging it.

**Comment by furqanmlk** [comment]
@lucassabreu 
https://automation-dashboard.vercel.app/cycles/55ccfcc7-9d0b-4444-ae75-a4a123c5fb38

All Tests are failing due to 
```
CypressError: cy.request() failed on:

http://server:8065/api/v4/users/email/sysadmin@sample.mattermost.com

The response we received from your web server was:

 401: Unauthorized
```

Can you run locally and see if it is failing ?

**Comment by lucassabreu** [comment]
> @lucassabreu https://automation-dashboard.vercel.app/cycles/55ccfcc7-9d0b-4444-ae75-a4a123c5fb38
> 
> All Tests are failing due to
> 
> ```
> CypressError: cy.request() failed on:
> 
> http://server:8065/api/v4/users/email/sysadmin@sample.mattermost.com
> 
> The response we received from your web server was:
> 
>  401: Unauthorized
> ```
> 
> Can you run locally and see if it is failing ?

only in the weekend now @furqanmlk

**Comment by lucassabreu** [comment]
hi @furqanmlk the error shown does not happen for me locally, i am merging master on the PR again to see if it helps the test server.

File: components/channel_info_rhs/channel_info_rhs.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/channel_info_rhs/channel_info_rhs.tsx:86]: this is actually a bug, something we should be handling better. Thank for investigating this.

Comment (M-ZubairAhmed) [reply components/channel_info_rhs/channel_info_rhs.tsx:86]: this is now handled in the code, you may revert this

Comment (lucassabreu) [reply components/channel_info_rhs/channel_info_rhs.tsx]: done

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.
